### ISSUE TITLE
problem with release 0.4

### ISSUE THREAD

**Issue by okwatts** [venomix666/nano6502#5]
when i boot the new image with the new version of the fs I get the CP/M-65 for nano6502 followed by "Couldn't open CCP"
Is this what you see?

**Comment by venomix666** [comment]
Hmm. No, for me it boots fine, but I saw that issue in some intermediate build. I'll look in to it.

**Comment by venomix666** [comment]
I've looked a bit at this, and for some reason which I don't understand yet some cpm65 builds doesn't want to boot with this version of the bitfile.

Can you try to use the nano6502_sysonly.img which is also included in the release, since that works fine for me? This will only overwrite your A-drive.

**Comment by okwatts** [comment]
Sorry same response, although with the first attempt it said "CP/M for nano6502" but then there was no response. Next time on reset it said couldn't open CCP.
Edit: Just tried with latest CP/M-65 image from David same problem. I note he doesn't have the bitfile in the latest release.

**Comment by venomix666** [comment]
I tried making a couple of alternative builds with different clock division for the SD-card interface, I added them to the [0.4 release](https://github.com/venomix666/nano6502/releases/tag/v0.4.0). 

The normal build uses a clock division of 3 and works well for me, could you check if a division of 2 or 4 works for you? It may be that some timing is on the edge and it becomes dependent on which SD-card you use or something like that.

Regarding the bitfile in CP/M-65 release that is expected. The CP/M-65 release only contains the CP/M-65 build and is autogenerated.

**Comment by okwatts** [comment]
That might be the reason. I tried the div2 and got the error "BDOS error: File is R/O". The file div4 gets me to the A> prompt and seems to work. Perhaps it is the response of the SD card

**Comment by okwatts** [comment]
I just tried the newest version(Oct30) of CP/M-65 from David's site and even with the div4 version it wouldn't start. Gives the signon header and then no response. Going back to your version 0.4 works fine. Is there more going  on than the SD card response time?

**Comment by venomix666** [comment]
Thanks for the feedback.

This looks very much like when a signal needs to be registered in FPGA but isn't, i.e. it changes and get read at the same time so the result becomes somewhat random and different between different individual FPGA chips etc.

I'll keep digging.

**Comment by venomix666** [comment]
I think I found it - I had to delay the write enable signal from the SD-card controller to the SD buffer RAM with one clock cycle. I tried running it on my board with different clock division settings and all of them works which indicates that the issue is hopefully solved.

I have made a release 0.4.1 which includes the fix. Please let me know if it works for you, and if so I'll close this issue.

**Comment by okwatts** [comment]
With the 041 .fs, I tried both your 041 img and David's 1101 img and both only give the signon but no response. Your img  0.4 does however boot to the a: prompt and I can assemble hello.asm. This is very peculiar.

**Comment by venomix666** [comment]
Yes, it is very strange. I spent a few hours trying to find the issue this weekend but only got more confused. Changing the memory mapping allowed for CP/M-65 seems to sort of move the issue around, so either it doesn't boot, either the assembler doesn't work or either the "free" commands hangs or produces strange output.

I'm leaning towards a timing issue still, but it may be that it somehow causes memory corruption in some edge cases. More investigating to be done...
